src/python/CRABClient/JobType/Analysis.py

```diff
@@ -148,7 +148,7 @@ def run(self, filecacheurl = None):
         run_ranges_is_valid = run_ranges is not None and isinstance(run_ranges, str) and re.match('^\d+((?!(-\d+-))(\,|\-)\d+)*$', run_ranges)
         if run_ranges_is_valid:
             run_list = getRunList(run_ranges)
-            if lumi_list:
+            if lumi_list != None:
                 lumi_list.selectRuns(run_list)
```

Comment:
Are you sure this works? I think the key is here. If lumi list and run range were both specified but have no overlap, then lumi_list should be empty. Maybe this works because {} (empty) is not None?